### PR TITLE
Move MetadataUrl concordium-std

### PR DESCRIPTION
**Pull request by annenkov**
## Purpose

Move MetadataUrl from CIS2 to `concordium-std` to make it available for various CIS standards

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-cis2/src/lib.rs [hunk 1]
```diff
@@ -62,30 +62,6 @@ pub const UPDATE_OPERATOR_EVENT_TAG: u8 = u8::MAX - 3;
...
 pub const TOKEN_METADATA_EVENT_TAG: u8 = u8::MAX - 4;
 
-/// Sha256 digest
-pub type Sha256 = [u8; 32];
-
-/// The location of the metadata and an optional hash of the content.
-// Note: For the serialization to be derived according to the CIS2
-// specification, the order of the fields cannot be changed.
-#[derive(Debug, Serialize, PartialEq, Eq, Clone)]
-pub struct MetadataUrl {
```
Comment (abizjak) [inline concordium-cis2/src/lib.rs:72]: Please also re-export MetadatUrl from cis2 so as to not break compatibility.

File: concordium-std/src/types.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ use crate::{
 use concordium_contracts_common::{AccountBalance, Amount, ParseError};
 use core::{fmt, str::FromStr};
 // Re-export for backward compatibility.
-pub use concordium_contracts_common::ExchangeRates;
+pub use concordium_contracts_common::*;
```
[the remark this answers is not in the document]
Comment (annenkov) [reply concordium-std/src/types.rs:7]: Yes, my bad, I changed it while experimenting. Will change back.

File: concordium-std/src/types.rs [hunk 3]
```diff
@@ -1247,3 +1247,12 @@ pub enum StateError {
...
     SubtreeWithPrefixNotFound,
 }
+
+/// The location of the metadata and an optional hash of the content.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct MetadataUrl {
+    /// The URL following the specification RFC1738.
+    pub url:  String,
+    /// A optional hash of the content.
+    pub hash: Option<HashSha2256>,
```
[the remark this answers is not in the document]
Comment (annenkov) [reply concordium-std/src/types.rs:1257]: > Is there a strong reason for making this breaking change?

My reasoning was that we already have this type, why introduce a new alias? But I don't mind keeping the alias. I'll move te alias to `concordium-std` then.

File: concordium-cis2/src/lib.rs (hunk 1 above)
Comment (annenkov) [reply concordium-cis2/src/lib.rs:72]: Done.

File: concordium-std/src/types.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (annenkov) [reply concordium-std/src/types.rs]: Fixed.

File: concordium-std/src/types.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (annenkov) [reply concordium-std/src/types.rs]: Keep byte array in `MetadataUrl`, use alias in wccd.

File: examples/cis2-wccd/src/lib.rs [hunk 4]
```diff
@@ -1165,10 +1161,10 @@ mod tests {
         // Set up crypto primitives to hash the document.
         let crypto_primitives = TestCryptoPrimitives::new();
         // The hash of the document stored at the above URL.
-        let initial_metadata_hash: Sha256 =
-            crypto_primitives.hash_sha2_256("document".as_bytes()).0;
+        let initial_metadata_hash: HashSha2256 =
+            HashSha2256(crypto_primitives.hash_sha2_256("document".as_bytes()).0);
```
[the remark this answers is not in the document]
Comment (annenkov) [reply examples/cis2-wccd/src/lib.rs]: This and other similar places are reverted to use a byte array.

**Pull request merged**